File: IT/1/Stringhe/Java/TestRegex.java
/* file TestRegex.java */

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class TestRegex {
	public static void main(String [] args) {
		/* prepara un'espressione regolare, e la compila per essere riutilizzata */

		String regex = "^Ul[iy](?:ss|x)es?$";
		Pattern pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);

		String [] s = {"Ulisse", "ulysse", "Ulysses", "ULIXES"};

		for (String name : s) {
			Matcher matcher = pattern.matcher(name);
			if (matcher.matches()) {
				System.out.println("La regex \"" + regex + "\" coincide con il nome \"" + name + "\".");
			}
		}
	}
}

File: IT/1/Stringhe/Python/test_regex.py
# file test_regex.py

from __future__ import print_function, unicode_literals
import re

# prepara un'espressione regolare, e la compila per essere riutilizzata

regex = r"^Ul[iy](?:ss|x)es?$"
pattern = re.compile(regex, re.IGNORECASE)

s = ["Ulisse", "ulysse", "Ulysses", "ULIXES"]

for name in s:
	if pattern.match(name): 
		print("La regex \"{}\" coincide con il nome \"{}\".".format(regex, name))

